cpp/src/arrow/ipc/adapter.h

```diff
@@ -78,6 +78,8 @@ class ARROW_EXPORT RowBatchReader {
   static Status Open(MemorySource* source, int64_t position, int max_recursion_depth,
       std::shared_ptr<RowBatchReader>* out);
 
+  ~RowBatchReader();
```

Comment:
Please make the constructor virtual. This is probably not the problem why Travis fails but at least a problem.